### PR TITLE
Orientation support

### PR DESCRIPTION
**Pull request by dabingnn**
add, sys, view, configuration to engine
sys: isMobile, browserType, browserVersion, webGLsupport
view: wrap htmlElements, support orientation
move input point calculation to view
configuration: just a interface
add config to playground, only works on mobile device

### CODE REVIEW

**Comment by dabingnn** [comment]
for #588 
for #371

File: lib/app.js [hunk 1]
```diff
@@ -4,6 +4,9 @@ import Input from './input';
 import { App as AppBase } from './ecs';
 import { EventEmitter } from './event-sys';
 import { utils as sceneUtils } from './scene-graph';
+import { vec2, vec3 } from './vmath';
+import { intersect } from './geom-utils';
```
Comment (YunHsiao) [inline lib/app.js]: redundant import?

**Comment by YunHsiao** [comment]
playground里有几个场景(frustum等)有直接改动`app._canvas.width`的操作，需要修改吗

**Comment by dabingnn** [comment]
app._canvas.width是需要调整的，现在在app中还保留的原因是兼容性，此外，希望暂时把app._canvas.width这些当作只读属性比较好

File: lib/app.js (hunk 1 above)
Comment (dabingnn) [reply lib/app.js]: thanks

**Pull request merged**